Tasks/Common/AzureRmDeploy-common/azure-arm-rest/azureModels.ts

```diff
@@ -205,7 +205,7 @@ export interface StorageAccount extends AzureBaseObject {
 }
 
 export interface AzureEndpoint {
-    subscriptionID: string;
+    subscriptionID?: string;
```

Comment:
why do we not have subscriptionid in case of publish profile?

Reply:
only subscription name and resourceId is present in the publish profile endpoint.

Reply:
but why not subscription Id? cant we add that